### PR TITLE
[Movie Gen] Performance optimization and other minor changes

### PR DESCRIPTION
**Pull request by hadipash**
# What does this PR do?

Fixes # (issue)

- Links in docs.

Adds # (feature)

- Switch FA layout from `BSND` to `BNSD` - improves training speed by ~10% (depending on a stage).
- Cache frequences in `TimestepEmbedder` as they're fixed.
- Drop unnecessary video tokens permutations.
- Sample video frames deterministically starting from 0 position (to better align with captions; can be enabled with `--dataset.deterministic_sample=True`).
- Adjust learning rates.
- Add 30B model Stage 3 training script.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

**Review by SamitHuang — APPROVED** [review]

File: examples/moviegen/mg/dataset/dataset.py [hunk 1]
```diff
@@ -164,18 +164,14 @@ def _get_item(self, idx: int, thw: Optional[Tuple[int, int, int]] = None) -> Tup
...
-            latent_mean, latent_std = tae_latent_data["latent_mean"], tae_latent_data["latent_std"]  # C T H W
-            if latent_mean.shape[1] < self._min_length:  # TODO: add support for images and buckets
+            latent_mean, latent_std = tae_latent_data["latent_mean"], tae_latent_data["latent_std"]  # T C H W
+            if 1 < len(latent_mean) < self._min_length:  # TODO: add support for buckets
                 raise ValueError(f"Video is too short: {data['video']}")
 
-            start_pos = random.randint(0, len(latent_mean) - self._min_length)
-            batch_index = np.linspace(start_pos, start_pos + self._min_length - 1, num_frames, dtype=int)
-
+            batch_index = np.linspace(0, self._min_length - 1, num_frames, dtype=int)
```
Comment (SamitHuang) [inline examples/moviegen/mg/dataset/dataset.py:171]: may loss diversity. better allow setting

**Review by CaitinZhao — APPROVED** [review]
